bots/archive/bot.py

```diff
@@ -0,0 +1,194 @@
+                raise Exception('Bucket name not set! Please define bucket name in project.settings')
+            self.bucket = get_bucket(settings.CELERY_TASK_BUCKET_NAME)
+
+        logger.info('%s started converting queryset to csv data', self.started_by)
+
+        self.chunk_size = chunk_size
+
+        current_time = datetime.datetime.utcnow()
+        one_day = datetime.timedelta(days=-1)
+        one_week = datetime.timedelta(weeks=-1)
+        two_weeks = datetime.timedelta(weeks=-2)
+
+        # bots.elasticsearch (24hrs)
+        elasticsearch_tasks = CeleryProviderTask.objects.filter(
+            app_label='elasticsearch',
+            timestamp__lt=current_time + one_day
+        )
+        # normalizertask (1 week)
+        normalizer_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.NormalizerTask',
+            timestamp__lt=current_time + one_week
+        )
+        # disambiguatortask (1 week)
+        disambiguator_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.DisambiguatorTask',
+            timestamp__lt=current_time + one_week
+        )
+        # harvestertask (2 weeks)
+        harvester_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.HarvesterTask',
+            timestamp__lt=current_time + two_weeks
+        )
+        # archivetask (2 weeks)
+        archive_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.ArchiveTask',
+            timestamp__lt=current_time + two_weeks
+        )
+
+        if elasticsearch_tasks.exists():
+            self.archive_queryset(
+                elasticsearch_tasks,
+                'elasticsearch/elasticsearch_tasks_',
+                'Elasticsearch CeleryTask'
+            )
+
+        if normalizer_tasks.exists():
+            self.archive_queryset(
+                normalizer_tasks,
+                'normalizer/normalizer_tasks_',
+                'NormalizerTask'
+            )
+
+        if disambiguator_tasks.exists():
+            self.archive_queryset(
+                disambiguator_tasks,
+                'disambiguator/disambiguator_tasks_',
+                'DisambiguatorTask'
+            )
+
+        if harvester_tasks.exists():
+            self.archive_queryset(
+                harvester_tasks,
+                'harvester/harvester_tasks_',
+                'HarvesterTask'
+            )
+
+        if archive_tasks.exists():
+            self.archive_queryset(
+                archive_tasks,
+                'archive/archive_tasks_',
+                'ArchiveTask'
+            )
+
+    def archive_queryset(self, queryset, location, task_name):
+        total = queryset.count()
+        logger.info('Found %s %ss eligible for archiving', total, task_name)
+        logger.info('Archiving in chunks of %d', self.chunk_size)
+
+        model = queryset.model
+        for i, celery_tasks in enumerate(chunk(queryset.all(), self.chunk_size)):
```

Comment:
Use `queryset.iterator` here, it'll save a lot of memory. `.all()` Loads everything into memory at once.